src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -118,8 +125,16 @@
   (when-not (string? (:path (:match-request rule)))
     (throw (IllegalArgumentException.
             (str "The path set in the authorization rule specified as "
-                 (pprint-rule (:path (:match-request rule))) " is invalid. "
+                 (pprint-rule rule) " is invalid. "
                  "It should be a string."))))
+  (when (and (not (nil? (:method rule)))
+             (nil? (some #(= % (:method rule)) valid-methods)))
+    (throw (IllegalArgumentException.
+             (str "The method specified in the authorization rule specified as "
+                  (pprint-rule rule) " is invalid. "
+                  "It should be either a string or list of strings that is equal "
+                  "to one of the following methods: 'get' 'post' 'put' 'delete' "
+                  "'head' 'any'"))))
```

Comment:
Yeah, I like the idea of keeping `any` as something we only manage internally to tk-authz.  The user can then get `any` behavior by just omitting `method` from a rule.  FWIW, that would seem to be consistent with how this was handled in legacy Ruby Puppet's auth.conf - https://docs.puppetlabs.com/puppet/latest/reference/config_file_auth.html#method.